// ngx-linkifyjs/esm5/ngx-linkifyjs.es5.js
import { CommonModule } from '@angular/common';
import { Injectable, Pipe, InjectionToken, NgModule, Inject } from '@angular/core';
import * as linkify from 'linkifyjs';
import { find, test } from 'linkifyjs';
import hashtag from 'linkifyjs/plugins/hashtag';
import mention from 'linkifyjs/plugins/mention';
import linkifyStr from 'linkifyjs/string';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var NgxLinkifyjsService = /** @class */ (function () {
    function NgxLinkifyjsService() {
    }
    /**
     * Convert the passed text as a string to an appropriate url
     *
     * @param text - the string to convert
     * @param options - options to pass it to the linkifyjs library
     */
    /**
     * Convert the passed text as a string to an appropriate url
     *
     * @param {?} text - the string to convert
     * @param {?=} options - options to pass it to the linkifyjs library
     * @return {?}
     */
    NgxLinkifyjsService.prototype.linkify = /**
     * Convert the passed text as a string to an appropriate url
     *
     * @param {?} text - the string to convert
     * @param {?=} options - options to pass it to the linkifyjs library
     * @return {?}
     */
    function (text, options) {
        // @ts-ignore
        return text.linkify(options);
    };
    /**
     * Find any links in a given text as a string
     *
     * @param text - the string to find some links
     */
    /**
     * Find any links in a given text as a string
     *
     * @param {?} text - the string to find some links
     * @return {?}
     */
    NgxLinkifyjsService.prototype.find = /**
     * Find any links in a given text as a string
     *
     * @param {?} text - the string to find some links
     * @return {?}
     */
    function (text) {
        return find(text);
    };
    /**
     * Test if a given value or array of values are links
     *
     * @param value - the value to test
     */
    /**
     * Test if a given value or array of values are links
     *
     * @param {?} value - the value to test
     * @return {?}
     */
    NgxLinkifyjsService.prototype.test = /**
     * Test if a given value or array of values are links
     *
     * @param {?} value - the value to test
     * @return {?}
     */
    function (value) {
        if (typeof value === 'string') {
            return test(value);
        }
        return test.apply(linkify, value);
    };
    NgxLinkifyjsService.decorators = [
        { type: Injectable },
    ];
    /** @nocollapse */
    NgxLinkifyjsService.ctorParameters = function () { return []; };
    return NgxLinkifyjsService;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var NgxLinkifyjsPipe = /** @class */ (function () {
    function NgxLinkifyjsPipe() {
    }
    /**
     * @param {?} value
     * @param {?=} options
     * @return {?}
     */
    NgxLinkifyjsPipe.prototype.transform = /**
     * @param {?} value
     * @param {?=} options
     * @return {?}
     */
    function (value, options) {
        return value ? linkifyStr(value, options) : value;
    };
    NgxLinkifyjsPipe.decorators = [
        { type: Pipe, args: [{
                    name: 'linkify'
                },] },
    ];
    return NgxLinkifyjsPipe;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/** @enum {string} */
var LinkType = {
    URL: 'url',
    HASHTAG: 'hashtag',
    MENTION: 'mention',
    EMAIL: 'email',
};

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/** @type {?} */
var NgxLinkifyjsConfigToken = new InjectionToken('NgxLinkifyjsConfig');
/** @type {?} */
var DEFAULT_CONFIG = { enableHash: true, enableMention: true };
var NgxLinkifyjsModule = /** @class */ (function () {
    function NgxLinkifyjsModule(config) {
        this.config = config;
        if (config.enableHash) {
            hashtag(linkify);
        }
        if (config.enableMention) {
            mention(linkify);
        }
    }
    /**
     * @param {?=} config
     * @return {?}
     */
    NgxLinkifyjsModule.forRoot = /**
     * @param {?=} config
     * @return {?}
     */
    function (config) {
        if (config === void 0) { config = DEFAULT_CONFIG; }
        return {
            ngModule: NgxLinkifyjsModule,
            providers: [
                NgxLinkifyjsService,
                {
                    provide: NgxLinkifyjsConfigToken,
                    useValue: config
                },
            ]
        };
    };
    NgxLinkifyjsModule.decorators = [
        { type: NgModule, args: [{
                    imports: [
                        CommonModule
                    ],
                    exports: [NgxLinkifyjsPipe],
                    declarations: [NgxLinkifyjsPipe]
                },] },
    ];
    /** @nocollapse */
    NgxLinkifyjsModule.ctorParameters = function () { return [
        { type: undefined, decorators: [{ type: Inject, args: [NgxLinkifyjsConfigToken,] }] }
    ]; };
    return NgxLinkifyjsModule;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */

export { LinkType, NgxLinkifyjsPipe, NgxLinkifyjsService, NgxLinkifyjsConfigToken, DEFAULT_CONFIG, NgxLinkifyjsModule };
