// elements-zone-strategy/esm5/lib/element-zone-strategy-factory.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { createCustomElement } from '@angular/elements';
import { ElementZoneStrategy } from './element-zone-strategy';
var ElementZoneStrategyFactory = /** @class */ (function () {
    function ElementZoneStrategyFactory(component, injector) {
        this.component = component;
        this.injector = injector;
        this.ngElement = createCustomElement(this.component, { injector: this.injector });
        // this is to keep the document-register-element polyfill happy
        // it doesn't like creating elements before the define, so we fill the registry with a random definition
        customElements.define(this.generateName(), this.ngElement);
    }
    /**
     * @return {?}
     */
    ElementZoneStrategyFactory.prototype.string4 = /**
     * @return {?}
     */
    function () {
        return Math.floor((1 + Math.random()) * 0x10000).toString(16).substring(1);
    };
    /**
     * @return {?}
     */
    ElementZoneStrategyFactory.prototype.generateName = /**
     * @return {?}
     */
    function () {
        var /** @type {?} */ result = 'dummy-name-n';
        var /** @type {?} */ temp = ElementZoneStrategyFactory.counter + '';
        result = result + temp + '-' + this.string4() + this.string4() + this.string4() + '-' +
            this.string4() + this.string4() + '-' + this.string4();
        ElementZoneStrategyFactory.counter++;
        return result;
    };
    /**
     * @param {?} injector
     * @return {?}
     */
    ElementZoneStrategyFactory.prototype.create = /**
     * @param {?} injector
     * @return {?}
     */
    function (injector) {
        // the only way to get a default strategy outside @angular/elements
        // is to create the ngElement/ngElementImpl and get it from a property
        var /** @type {?} */ tempElement = new this.ngElement(this.injector);
        var /** @type {?} */ strategy = tempElement['ngElementStrategy'];
        tempElement = null;
        return new ElementZoneStrategy(strategy, this.injector);
    };
    ElementZoneStrategyFactory.counter = 1;
    return ElementZoneStrategyFactory;
}());
export { ElementZoneStrategyFactory };
function ElementZoneStrategyFactory_tsickle_Closure_declarations() {
    /** @type {?} */
    ElementZoneStrategyFactory.counter;
    /** @type {?} */
    ElementZoneStrategyFactory.prototype.ngElement;
    /** @type {?} */
    ElementZoneStrategyFactory.prototype.component;
    /** @type {?} */
    ElementZoneStrategyFactory.prototype.injector;
}
