// elements-zone-strategy/esm5/lib/element-zone-strategy.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { NgZone } from '@angular/core';
var ElementZoneStrategy = /** @class */ (function () {
    function ElementZoneStrategy(strategy, injector) {
        this.strategy = strategy;
        this.injector = injector;
        this.ngZone = this.injector.get(NgZone);
    }
    /**
     * @param {?} element
     * @return {?}
     */
    ElementZoneStrategy.prototype.connect = /**
     * @param {?} element
     * @return {?}
     */
    function (element) {
        var _this = this;
        this.runInZone(function () { _this.strategy.connect(element); });
        this.events = this.strategy.events;
    };
    /**
     * @return {?}
     */
    ElementZoneStrategy.prototype.disconnect = /**
     * @return {?}
     */
    function () {
        var _this = this;
        this.runInZone(function () { _this.strategy.disconnect(); });
    };
    /**
     * @param {?} propName
     * @return {?}
     */
    ElementZoneStrategy.prototype.getInputValue = /**
     * @param {?} propName
     * @return {?}
     */
    function (propName) {
        var _this = this;
        // tslint:disable-next-line:arrow-return-shorthand
        return this.runInZone(function () { return _this.strategy.getInputValue(propName); });
    };
    /**
     * @param {?} propName
     * @param {?} value
     * @return {?}
     */
    ElementZoneStrategy.prototype.setInputValue = /**
     * @param {?} propName
     * @param {?} value
     * @return {?}
     */
    function (propName, value) {
        var _this = this;
        this.runInZone(function () { _this.strategy.setInputValue(propName, value); });
    };
    /**
     * @param {?} fn
     * @return {?}
     */
    ElementZoneStrategy.prototype.runInZone = /**
     * @param {?} fn
     * @return {?}
     */
    function (fn) { return this.ngZone.run(fn); };
    return ElementZoneStrategy;
}());
export { ElementZoneStrategy };
function ElementZoneStrategy_tsickle_Closure_declarations() {
    /** @type {?} */
    ElementZoneStrategy.prototype.ngZone;
    /** @type {?} */
    ElementZoneStrategy.prototype.events;
    /** @type {?} */
    ElementZoneStrategy.prototype.strategy;
    /** @type {?} */
    ElementZoneStrategy.prototype.injector;
}
