// elements-zone-strategy/esm2015/lib/element-zone-strategy.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { NgZone } from '@angular/core';
export class ElementZoneStrategy {
    /**
     * @param {?} strategy
     * @param {?} injector
     */
    constructor(strategy, injector) {
        this.strategy = strategy;
        this.injector = injector;
        this.ngZone = this.injector.get(NgZone);
    }
    /**
     * @param {?} element
     * @return {?}
     */
    connect(element) {
        this.runInZone(() => { this.strategy.connect(element); });
        this.events = this.strategy.events;
    }
    /**
     * @return {?}
     */
    disconnect() {
        this.runInZone(() => { this.strategy.disconnect(); });
    }
    /**
     * @param {?} propName
     * @return {?}
     */
    getInputValue(propName) {
        // tslint:disable-next-line:arrow-return-shorthand
        return this.runInZone(() => { return this.strategy.getInputValue(propName); });
    }
    /**
     * @param {?} propName
     * @param {?} value
     * @return {?}
     */
    setInputValue(propName, value) {
        this.runInZone(() => { this.strategy.setInputValue(propName, value); });
    }
    /**
     * @param {?} fn
     * @return {?}
     */
    runInZone(fn) { return this.ngZone.run(fn); }
}
function ElementZoneStrategy_tsickle_Closure_declarations() {
    /** @type {?} */
    ElementZoneStrategy.prototype.ngZone;
    /** @type {?} */
    ElementZoneStrategy.prototype.events;
    /** @type {?} */
    ElementZoneStrategy.prototype.strategy;
    /** @type {?} */
    ElementZoneStrategy.prototype.injector;
}
