// elements-zone-strategy/esm2015/lib/element-zone-strategy-factory.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { createCustomElement } from '@angular/elements';
import { ElementZoneStrategy } from './element-zone-strategy';
export class ElementZoneStrategyFactory {
    /**
     * @param {?} component
     * @param {?} injector
     */
    constructor(component, injector) {
        this.component = component;
        this.injector = injector;
        this.ngElement = createCustomElement(this.component, { injector: this.injector });
        // this is to keep the document-register-element polyfill happy
        // it doesn't like creating elements before the define, so we fill the registry with a random definition
        customElements.define(this.generateName(), this.ngElement);
    }
    /**
     * @return {?}
     */
    string4() {
        return Math.floor((1 + Math.random()) * 0x10000).toString(16).substring(1);
    }
    /**
     * @return {?}
     */
    generateName() {
        let /** @type {?} */ result = 'dummy-name-n';
        const /** @type {?} */ temp = ElementZoneStrategyFactory.counter + '';
        result = result + temp + '-' + this.string4() + this.string4() + this.string4() + '-' +
            this.string4() + this.string4() + '-' + this.string4();
        ElementZoneStrategyFactory.counter++;
        return result;
    }
    /**
     * @param {?} injector
     * @return {?}
     */
    create(injector) {
        // the only way to get a default strategy outside @angular/elements
        // is to create the ngElement/ngElementImpl and get it from a property
        let /** @type {?} */ tempElement = new this.ngElement(this.injector);
        const /** @type {?} */ strategy = tempElement['ngElementStrategy'];
        tempElement = null;
        return new ElementZoneStrategy(strategy, this.injector);
    }
}
ElementZoneStrategyFactory.counter = 1;
function ElementZoneStrategyFactory_tsickle_Closure_declarations() {
    /** @type {?} */
    ElementZoneStrategyFactory.counter;
    /** @type {?} */
    ElementZoneStrategyFactory.prototype.ngElement;
    /** @type {?} */
    ElementZoneStrategyFactory.prototype.component;
    /** @type {?} */
    ElementZoneStrategyFactory.prototype.injector;
}
